core/shared/src/main/scala/fs2/Segment.scala

```diff
@@ -835,6 +835,27 @@ abstract class Segment[+O,+R] { self =>
   }
 
   /**
+   * Converts this segment to an array, discarding the result.
+   *
```

Comment:
A `Segment[O,R]` outputs values of type `O` and when it finishes, outputs a single result of type `R`. Often `R` is `Unit`. When converting a segment to a standard collection type, we collect all the output `O` values and throw away the `R` value.